lazyflow/operators/generic.py

```diff
@@ -535,19 +535,13 @@ def execute(self, slot, subindex, roi, result):
         key = roi.toSlice()
         data = self.inputs["Input"][key[:-1] + (slice(None),)].wait()
 
-        dm = numpy.max(data, axis=data.ndim - 1, keepdims=True)
-        res = numpy.zeros(data.shape, numpy.uint8)
-        numpy.equal(data, dm, out=res)
-
-        # Special case: If all channels are tied, then none of them win.
-        tied_pixels = numpy.logical_and.reduce(res, axis=-1, keepdims=True, dtype=numpy.uint8)
+        # special case, when data is all zeros (e.g. directly from frozen cache w/o trained classifier)
```

Comment:
Is it significantly faster compared to the argmax branch?

Reply:
problem is the argmax branch will return `1`s, which will be rendered as solid color

Reply:
That's right, sorry, got confused by the previous "not fast enough" comments.

But what if the input just happens to be all zeroes?

Also, what if, later, the input changes from all-zeroes to a single non-zero pixel? Suddenly the output jumps from all-zeroes to almost-all-ones.

Reply:
when there are probabilities we assume they sum to 1. So if there is all zeros, we draw from cache. If there is one non-zero value, you can assume you have probabilities.